src/webgl/p5.RendererGL.js

```diff
@@ -1024,12 +1024,23 @@ class RendererGL extends Renderer {
     this.clear(..._col._getRGBA());
   }
 
-  // Combines the model and view matrices to get the uMVMatrix
-  // This method will be reusable wherever you need to update the combined matrix.
-  calculateCombinedMatrix() {
+
+  /**
+   * Get a matrix from world-space to screen-space
+   */
+  getWorldToScreenMatrix() {
     const modelMatrix = this.states.uModelMatrix;
     const viewMatrix = this.states.uViewMatrix;
-    return modelMatrix.copy().mult(viewMatrix);
+    const projectionMatrix = this.states.uPMatrix;
+    const projectedToScreenMatrix = new Matrix(4);
+    projectedToScreenMatrix.scale(this.width, this.height, 1);
+    projectedToScreenMatrix.translate([0.5, 0.5, 0.5]);
+    projectedToScreenMatrix.scale(0.5, -0.5, 0.5);
+
+    const modelViewMatrix = modelMatrix.copy().mult(viewMatrix);
+    const modelViewProjectionMatrix = modelViewMatrix.mult(projectionMatrix);
+    const worldToScreenMatrix = modelViewProjectionMatrix.mult(projectedToScreenMatrix);
+    return worldToScreenMatrix;
```

Comment:
Tips on avoiding allocations here welcome :sweat_smile: I guess the `copy()` is unavoidable, but allocating `projectedToScreenMatrix` feels avoidable :thinking: (Or should I just cache it on resize?)

Reply:
Yeah, that's what I ended up discovering as well :sweat_smile: 
Let's just leave it as is for now; we need more sketches that use it before it's clear how impactful that one function is. And given that `modelMatrix.copy().mult(viewMatrix)` already happens multiple times per shape, I'm pretty sure there are slower things around anyway.

Reply:
ok sounds good, I think we're all set to merge this then! thanks for your work!